swarm/network/stream/delivery.go

```diff
@@ -208,6 +211,9 @@ R:
 		chunk.SData = req.SData
 		d.db.Put(chunk)
 		chunk.WaitToStore()
+		if chunk.GetErrored() == storage.ChunkErrInvalid {
+			req.peer.Drop(errors.New("invalid chunk"))
```

Comment:
I think we should log a Warn when we drop peers, and it's not necessarily an `error` (which I haven't assessed for this particular case)

Reply:
@zelig wanted to drop the connection on invalid chunks.